src/events/SDL_touch.c

```diff
@@ -255,7 +255,7 @@ SDL_SendTouch(SDL_TouchID id, SDL_FingerID fingerid, SDL_Window * window,
 #if SYNTHESIZE_TOUCH_TO_MOUSE
     /* SDL_HINT_TOUCH_MOUSE_EVENTS: controlling whether touch events should generate synthetic mouse events */
     {
-        if (mouse->touch_mouse_events) {
+        if (mouse->touch_mouse_events && (!mouse->touch_mouse_events_direct_only || touch->type == SDL_TOUCH_DEVICE_DIRECT) ) {
             /* FIXME: maybe we should only restrict to a few SDL_TouchDeviceType */
```

Comment:
I think this FIXME could be removed now.

Reply:
Done. I've also added comment similar to one about SDL_HINT_TOUCH_MOUSE_EVENTS